scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -351,26 +353,90 @@ private void initFooter() {
+
+        Document dom = XMLParser.parse(jobDescriptor);
+        String documentation = null;
+        String icon = null;
+        String bucketName = null;
+        Boolean existBucketName = false;
+        Boolean existDocumentation = false;
+        Boolean existIcon = false;
+
+        NodeList genericInfo = dom.getElementsByTagName("genericInformation");
+        // check if the job has genericInformation or not
+        if (genericInfo != null && genericInfo.getLength() > 0) {
+            // get the first item
+            Node root = genericInfo.item(0);
+            NodeList list = root.getChildNodes();
+            for (int i = 0; i < list.getLength(); i++) {
+                Node node = list.item(i);
+                if (node.getNodeName().equals("info") && node.hasAttributes()) {
+                    NamedNodeMap attributes = node.getAttributes();
+                    for (int j = 0; j < attributes.getLength(); j++) {
+                        Node attribute = attributes.item(j);
+
+                        if (attribute.getNodeType() == Node.ATTRIBUTE_NODE && attribute.getNodeName().equals("name") &&
+                            attribute.getNodeValue().equalsIgnoreCase("bucketName")) {
+                            existBucketName = true;
+                        }
+                        if (attribute.getNodeType() == Node.ATTRIBUTE_NODE && attribute.getNodeName().equals("name") &&
+                            attribute.getNodeValue().equalsIgnoreCase("documentation")) {
+                            existDocumentation = true;
+                        }
+                        if (attribute.getNodeType() == Node.ATTRIBUTE_NODE && attribute.getNodeName().equals("name") &&
+                            attribute.getNodeValue().equalsIgnoreCase("workflow.icon")) {
+                            existIcon = true;
+                        }
+                        if (attribute.getNodeType() == Node.ATTRIBUTE_NODE && attribute.getNodeName().equals("value") &&
+                            existBucketName) {
+                            bucketName = attribute.getNodeValue();
+                            existBucketName = false;
+                        }
+                        if (attribute.getNodeType() == Node.ATTRIBUTE_NODE && attribute.getNodeName().equals("value") &&
+                            existDocumentation) {
+                            documentation = attribute.getNodeValue();
+                            existDocumentation = false;
+                        }
+                        if (attribute.getNodeType() == Node.ATTRIBUTE_NODE && attribute.getNodeName().equals("value") &&
+                            existIcon) {
+                            icon = attribute.getNodeValue();
+                            existIcon = false;
+                        }
+                    }
+                }
+            }
+        }
+
+        Widget worfklowMetaDataWidget = new HTML("<center> <img src=" + icon +
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "worfklowMetaDataWidget". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)